src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -2100,6 +2100,32 @@ theorem toNat_abs {x : BitVec w} : x.abs.toNat = if x.msb then 2^w - x.toNat els
     rw [Nat.mod_eq_of_lt (by omega)]
   · simp [h]
 
+theorem getLsbD_abs {i : Nat} {x : BitVec w} :
+   getLsbD x.abs i = if x.msb then getLsbD (-x) i else getLsbD x i := by
+  by_cases h : x.msb <;> simp [BitVec.abs, h]
+
+theorem getMsbD_abs {i : Nat} {x : BitVec w} :
+    getMsbD (x.abs) i = if x.msb then getMsbD (-x) i else getMsbD x i := by
+  by_cases h : x.msb <;> simp [BitVec.abs, h]
+
+theorem msb_abs {w : Nat} {x : BitVec w} :
+    (x.abs).msb = if x.msb then (-x).msb else x.msb := by
+  by_cases h : x.msb <;> simp [BitVec.abs, h]
+
+/-! ### neg -/
+
+theorem msb_neg {x : BitVec w} :
+    (-x).msb = (~~~x + 1#w).msb := by
+  rw [neg_eq_not_add]
```

Comment:
```suggestion
theorem msb_neg {x : BitVec w} :
    (-x).msb = !x.msb := by
```

Here, too: the sign bit of `-x` is always the inverse of the signbit of `x`.

You might want to pull this result up higher in the file, as it's useful to prove `msb_abs`

Reply:
There is one exception:

```lean4
@[simp]
theorem neg_intMin {w : Nat} : -intMin w = intMin w := by
  by_cases h : 0 < w
  · simp [bv_toNat, h]
  · simp only [Nat.not_lt, Nat.le_zero_eq] at h
    simp [bv_toNat, h]
```

https://github.com/leanprover/lean4/blob/master/src/Init/Data/BitVec/Lemmas.lean#L2841

We may want to first pull-up `intMin` and `intMax` before add/sub/neg, and then use those.

Reply:
thank you! I am working on the proof for this `msb_neg`, will draft the pr for the time being